src/main/java/net/maritimeconnectivity/serviceregistry/models/dto/InstanceDto.java

```diff
@@ -64,18 +62,18 @@ public class InstanceDto implements Serializable, JsonSerializable {
     private String organizationId; // Use the JWT auth token for that
     private List<String> unlocode;
     private String endpointUri;
-    private String endpointType;
+    private String statusEndpointUri;
     private String mmsi;
     private String imo;
-    private List<String> serviceType;
+    private List<String> serviceTypes;
     private List<SECOM_DataProductType> dataProductType;
     private XmlDto instanceAsXml;
     private DocDto instanceAsDoc;
     private Long ledgerRequestId;
     private LedgerRequestStatus ledgerRequestStatus;
     private Set<Long> docIds = new HashSet<>();
-    private String implementsServiceDesign;
-    private String implementsServiceDesignVersion;
+    private Map<String, String> implementsServiceDesigns;
+    private Map<String, String> designsServiceSpecifications;
```

Comment:
Can see that these are also defined as Maps in the Instance class, but I can't figure what they are supposed to map from and to. Would it be possible to use a (Hash)Set instead.

Reply:
Map between MRN of design/service-spec and version